compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2349,6 +2349,7 @@ public void visit(BLangIf ifNode) {
         }
 
         Map<BVarSymbol, BType.NarrowedTypes> prevNarrowedTypeInfo = this.narrowedTypeInfo;
+        Map<BVarSymbol, BType.NarrowedTypes> narrowedTypes = new HashMap<>();
```

Comment:
Can we use a name here that reflects what this is used for? Along with a comment maybe?